packages/tooling/__tests__/lib/flatten-schema.test.js

```diff
@@ -87,3 +88,166 @@ describe('$ref usages', () => {
     expect(flattenSchema(schema, oas)).toStrictEqual(expected);
   });
 });
+
+describe('polymorphism cases', () => {
+  describe('allOf', () => {
+    it('should flatten a schema to an array', () => {
+      const schema = {
+        type: 'array',
+        items: {
+          $ref: '#/components/schemas/Pet',
+        },
+      };
+
+      expect(flattenSchema(schema, petstoreExpanded)).toStrictEqual([
+        { name: 'name', type: 'String', description: undefined },
+        { name: 'tag', type: 'String', description: undefined },
+        { name: 'id', type: 'Integer', description: undefined },
+      ]);
+    });
+
+    it('should flatten a component schema to an array', () => {
+      const schema = petstoreExpanded.components.schemas.Pet;
+
+      expect(flattenSchema(schema, petstoreExpanded)).toStrictEqual([
+        { name: 'name', type: 'String', description: undefined },
+        { name: 'tag', type: 'String', description: undefined },
+        { name: 'id', type: 'Integer', description: undefined },
+      ]);
+    });
+
+    it('should be able to handle an allOf that contains deep $refs', () => {
+      const schema = {
+        allOf: [
+          {
+            $ref: '#/components/schemas/NewPet',
+          },
+          {
+            type: 'object',
+            required: ['id'],
+            properties: {
+              id: {
+                type: 'integer',
+                format: 'int64',
+              },
+            },
+          },
+        ],
+      };
+
+      const newPetSchema = petstore.components.schemas.Pet;
+      delete newPetSchema.properties.id;
+
+      expect(
+        flattenSchema(schema, {
+          components: {
+            schemas: {
+              Category: petstore.components.schemas.Category,
+              NewPet: newPetSchema,
+              Tag: petstore.components.schemas.Tag,
+            },
+          },
+        })
+      ).toStrictEqual([
+        { name: 'category', type: 'Object', description: undefined },
+        { name: 'category.id', type: 'Integer', description: undefined },
+        { name: 'category.name', type: 'String', description: undefined },
+        { name: 'name', type: 'String', description: undefined },
+        { name: 'photoUrls', type: '[String]', description: undefined },
+        { name: 'tags', type: '[Object]', description: undefined },
+        { name: 'tags[].id', type: 'Integer', description: undefined },
+        { name: 'tags[].name', type: 'String', description: undefined },
+        { name: 'status', type: 'String', description: 'pet status in the store' },
+        { name: 'id', type: 'Integer', description: undefined },
+      ]);
+    });
+  });
+
+  describe('anyOf', () => {
+    it('should flatten only the first schema listed', () => {
```

Comment:
This is good for now! We can work on this in future to allow selection in the UI.